iac/provider-gcp/nomad-cluster/scripts/start-client.sh

```diff
@@ -260,13 +267,36 @@ overcommitment_hugepages=$(remove_decimal $overcommitment_hugepages)
 echo "- Allocating $overcommitment_hugepages huge pages ($overcommitment_hugepages_percentage%) for overcommitment"
 echo $overcommitment_hugepages >/proc/sys/vm/nr_overcommit_hugepages
 
+# Get GCE DNS server dynamically from metadata for Consul recursors
+# This ensures we can resolve internet domains through Consul
+GCE_DNS=$(curl -s -H 'Metadata-Flavor: Google' http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/dns-servers || echo "169.254.169.254")
+
+# Start Consul first (in background) with GCE DNS as recursor
+# This allows Consul to handle both .consul queries AND forward internet queries
 # These variables are passed in via Terraform template interpolation
 /opt/consul/bin/run-consul.sh --client \
     --consul-token "${CONSUL_TOKEN}" \
     --cluster-tag-name "${CLUSTER_TAG_NAME}" \
     --enable-gossip-encryption \
     --gossip-encryption-key "${CONSUL_GOSSIP_ENCRYPTION_KEY}" \
-    --dns-request-token "${CONSUL_DNS_REQUEST_TOKEN}" &
+    --dns-request-token "${CONSUL_DNS_REQUEST_TOKEN}" \
+    --recursor "$${GCE_DNS}" &
+
+# Give Consul a moment to start its DNS server on port 8600
+sleep 3
```

Comment:
Please put some loop which will check if the server is reachable. Otherwise it's race-y and this will cause problems as we've seen before. The script should fail if the port 8600 isn't accessible in some timeout, that will prevent nomad to start and it will mark VM unhealthy after a while